.buildkite/scripts/steps/integration_tests_tf.sh

```diff
@@ -2,20 +2,63 @@
 set -euo pipefail
 
 source .buildkite/scripts/common2.sh
+
 source .buildkite/scripts/steps/ess.sh
 
+# Make sure that all tools are installed
+asdf install
+
+GROUP_NAME=$1
+TEST_SUDO=$2
+if [ -z "$GROUP_NAME" ]; then
+  echo "Error: Specify the group name: integration_tests_tf.sh [group_name]" >&2
+  exit 1
+fi
+
+if [ -z "$TEST_SUDO" ]; then
+  echo "Error: Specify the test sudo: integration_tests_tf.sh [group_name] [test_sudo]" >&2
+  exit 1
+fi
 
 # Override the agent package version using a string with format <major>.<minor>.<patch>
 # There is a time when the snapshot is not built yet, so we cannot use the latest version automatically
 # This file is managed by an automation (mage integration:UpdateAgentPackageVersion) that check if the snapshot is ready.
-OVERRIDE_AGENT_PACKAGE_VERSION="$(cat .package-version)"
-OVERRIDE_TEST_AGENT_VERSION=${OVERRIDE_AGENT_PACKAGE_VERSION}"-SNAPSHOT"
+OVERRIDE_STACK_VERSION="$(cat .package-version)"
+OVERRIDE_STACK_VERSION=${OVERRIDE_STACK_VERSION}"-SNAPSHOT"
 
-echo "~~~ Bulding test binaries"
+echo "~~~ Building test binaries"
 mage build:testBinaries
 
-ess_up $OVERRIDE_TEST_AGENT_VERSION || echo "Failed to start ESS stack" >&2
-trap 'ess_down' EXIT
+# If the step is retried, we start the stack again.
+# BUILDKITE_RETRY_COUNT == "0" for the first run
+# BUILDKITE_RETRY_COUNT > 0 for the retries
+if [[ "${BUILDKITE_RETRY_COUNT}" -gt 0 ]]; then
+  echo "~~~ The steps is retried, starting the ESS stack again"
```

Comment:
Can you confirm if `buildkite-agent meta-data get` can be used for secret access?

Reply:
The management of metadata and secrets differs from that of GH Actions. It's like a cross-step key-value storage with a job scope. You have access to what you put there. You can potentially get a secret from Vault and put it in metadata. We don't do that because it might cause a leak.

Reply:
> You can potentially get a secret from Vault and put it in metadata.  We don't do that because it might cause a leak.

IIUC, https://github.com/elastic/elastic-agent/pull/5113 does put a secret in metadata.

Reply:
Raised a follow-up issue